source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -170,16 +170,19 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
                                                                          context_.listenerScope())),
       proxy_100_continue_(config.proxy_100_continue()),
       delayed_close_timeout_(PROTOBUF_GET_MS_OR_DEFAULT(config, delayed_close_timeout, 1000)),
+#ifdef ENVOY_NORMALIZE_PATH_BY_DEFAULT
```

Comment:
Why this change?

Reply:
Can you share a citation to the standard for my benefit? Up until this point, I was under the impression that the preprocessor was a purely lexical step that occurred prior to any parsing or consideration of the language syntax, but I prefer to consume my standards lazily, they don't make great beach reading :)

Reply:
[Found the citation;  ISO/IEC 9899:2011 6.10.3 
```
The sequence of preprocessing tokens bounded by the outside-most matching parentheses forms the list of arguments for the function-like macro. The individual arguments within the list are separated by comma preprocessing tokens, but comma preprocessing tokens between matching inner parentheses do not separate arguments. If there are sequences of preprocessing tokens within the list of arguments that would otherwise act as preprocessing directives, the behavior is undefined.
```
The C Preprocessor is a serial line-oriented grammar, with one well understood line break exception sequence of '\' <newline> (which is evaluated as a single space.)
```
#define sum_values(arg1, arg2) ( (arg1 + arg2) )
int next(int value)
{
    /* The C preprocessor is expected to replace comments with a space
     * and smash whitespace while performing the macro expansion, so...
     */
    return sum_values(value,
#ifdef TWO_IF_BY_SEA
                      2
#else
                      1
#endif
    );
}
```
On some compilers, the #if expression is evaluated, in spite of interrupting macro evaluation and expansion.

On other compilers, the macro invocation is evaluated as;
```
    return ( (value) + (#ifdef TWO_IF_BY_SEA 2 #else 1 #endif) );
```
I was able to track down the details of this specific issue with some help from CMU's SEI CERT C Coding Standard, lots more great beach reading for you :)
https://wiki.sei.cmu.edu/confluence/display/c/CC.+Undefined+Behavior#CC.UndefinedBehavior-ub_93

Reply:
Ack, thanks for the explanation.